cmd/controller/controller.go

```diff
@@ -99,11 +106,212 @@ func NewController(store *store.AuthStore) *Controller {
 
 func (c *Controller) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	c.log.Info("Reconciling for", "name", req.NamespacedName.Name)
+
+	fromToForKey := req.NamespacedName.Namespace
+
+	crcSubjects := []v1a1.Subject{}
+	crcList := &v1a1.ClusterReferenceConsumerList{}
+	err := c.crClient.List(ctx, crcList)
+	if err != nil {
+		c.log.Error(err, "could not list ClusterReferenceConsumers")
+		return ctrl.Result{}, err
+	}
+
+	for _, crc := range crcList.Items {
+		for _, ref := range crc.References {
+			origin := fmt.Sprintf("%s/%s", ref.From.Group, ref.From.Resource)
+			target := fmt.Sprintf("%s/%s", ref.To.Group, ref.To.Resource)
+			key := fmt.Sprintf("%s;%s;%s", origin, target, ref.For)
+			if key == fromToForKey {
+				crcSubjects = append(crcSubjects, crc.Subject)
+				break
+			}
+		}
+	}
+
+	referencePaths := []*v1a1.ReferencePath{}
+	crgList := &v1a1.ClusterReferenceGrantList{}
+	err = c.crClient.List(ctx, crgList)
+	if err != nil {
+		c.log.Error(err, "could not list ClusterReferenceGrants")
+		return ctrl.Result{}, err
+	}
+
+	var fromVersion string
+
+	for _, crg := range crgList.Items {
+		origin := fmt.Sprintf("%s/%s", crg.From.Group, crg.From.Resource)
+		// Early exit if origin is not the same
+		if strings.Split(fromToForKey, ";")[0] != origin {
+			continue
+		}
+		if len(crg.Versions) == 0 {
+			continue
+		}
```

Comment:
Nit: Probably want to have log level + log these exits at something like debug level so we can just change the log level to figure out what's going on.

Reply:
added an info log